File: android/lib/src/main/cpp/Contact_Factory.cpp
//
//  Elastos.SDK.Contact.cpp
//
//  Created by mengxk on 19/03/16.
//  Copyright © 2016 mengxk. All rights reserved.
//

#include <Contact_Factory.hpp>

#include <Elastos.SDK.Contact.hpp>

/***********************************************/
/***** static variables initialize *************/
/***********************************************/

/***********************************************/
/***** static function implement ***************/
/***********************************************/

void Contact_Factory::SetLogLevel(int level)
{
    elastos::Contact::Factory::SetLogLevel(level);

    return;
}

int Contact_Factory::SetLocalDataDir(const std::string& dir)
{
    int ret = elastos::Contact::Factory::SetLocalDataDir(dir);

    return ret;
}

/***********************************************/
/***** class public function implement  ********/
/***********************************************/

/***********************************************/
/***** class protected function implement  *****/
/***********************************************/


/***********************************************/
/***** class private function implement  *******/
/***********************************************/

File: ios/sdk/swift/internal/ContactFactory.swift
//
//  ContactFactory.swift
//  sdk
//
//  Created by mengxk on 2019/9/19.
//  Copyright © 2019 Elastos. All rights reserved.
//

import CrossPL

/* @CrossClass */
class ContactFactory: CrossBase {
  /* @CrossNativeInterface */
  public static func SetLogLevel(level: Int32) {
    crosspl_Proxy_ContactFactory_SetLogLevel(level)
  }
  
  /* @CrossNativeInterface */
  public static func SetDeviceId(devId: String) {
    crosspl_Proxy_ContactFactory_SetDeviceId(devId)
  }
  
  /* @CrossNativeInterface */
  public static func SetLocalDataDir(dir: String) -> Int32 {
    return crosspl_Proxy_ContactFactory_SetLocalDataDir(dir)
  }
  
  init() {
    super.init(className: String(describing: ContactFactory.self))
 }
} // class Factory

